pkg/stub/handler.go

```diff
@@ -88,16 +95,77 @@ type Handler struct {
 
 	skippedTemplates    map[string]bool
 	skippedImagestreams map[string]bool
+
+	defaultResourceTimer *time.Timer
 }
 
-func (h *Handler) manageDockerCfgSecret(deleted bool, operatorCfg *v1alpha1.SamplesResource, secret *corev1.Secret) error {
-	// clear out temporary secret cache
-	h.registrySecret = corev1.Secret{}
+func (h *Handler) GoodStatusUpdate(srcfg *v1alpha1.SamplesResource, newStatus corev1.ConditionStatus) error {
+	status := srcfg.Condition(v1alpha1.SamplesExist)
+	// decision was made to not spam master if
+	// duplicate events come it (i.e. status does not
+	// change)
+	if status.Status != newStatus {
+		now := kapis.Now()
+		status.LastUpdateTime = now
+		status.Status = newStatus
+		status.LastTransitionTime = now
+		srcfg.ConditionUpdate(status)
+		err := h.sdkwrapper.Update(srcfg)
+		if err != nil {
+			if !kerrors.IsConflict(err) {
+				return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "failed adding success condition to config: %v")
+			}
+			logrus.Printf("got conflict error %#v on success status update, just going to forgo this time vs. retry", err)
+		}
+
+		logrus.Println("")
+		logrus.Println("")
+		logrus.Println("")
+		logrus.Println("")
+	}
+	h.samplesResource = srcfg
+	return nil
+}
+
+func (h *Handler) CreateDefaultResourceIfNeeded() error {
+	if h.samplesResource == nil {
```

Comment:
feels racy, do we need a mutex around references to this?

Reply:
sure will add mutex